src/expr/lazy_proof.cpp

```diff
@@ -56,36 +57,41 @@ std::shared_ptr<ProofNode> LazyCDProof::getProofFor(Node fact)
     if (it == visited.end())
     {
       visited.insert(cur);
-      if (cur->getRule() == PfRule::ASSUME)
+      Node cfact = cur->getResult();
+      if (getProof(cfact).get() != cur)
+      {
+        // we don't own this proof, skip it
```

Comment:
Clarify saying that this is so that the method is idempotent and does not expand assumptions that were added from previous expansions of generators.